bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/api/dto/clip2/Power.java

```diff
@@ -27,15 +28,18 @@
  */
 @NonNullByDefault
 public class Power {
-    private @NonNullByDefault({}) @SerializedName("battery_state") String batteryState;
+    private @Nullable @SerializedName("battery_state") String batteryState;
     private @SerializedName("battery_level") int batteryLevel;
 
     public BatteryStateType getBatteryState() {
-        try {
-            return BatteryStateType.valueOf(batteryState.toUpperCase());
-        } catch (IllegalArgumentException e) {
-            return BatteryStateType.CRITICAL;
+        String batteryState = this.batteryState;
+        if (batteryState != null) {
+            try {
+                return BatteryStateType.valueOf(batteryState.toUpperCase());
+            } catch (IllegalArgumentException e) {
+            }
         }
+        return BatteryStateType.CRITICAL;
```

Comment:
Not sure if reporting a critical level is the right thing for an unknown state. This might cause some negative sideeffects. Would it be possible to extend the enum with an  `Unknown`  state?

Reply:
Oups, sorry @lsiepel , you submitted you comment at the same time I approved.
But your question is good, I asked me the same.
As I understood the description, it should happen only during a short time.

Reply:
Waiting for @andrewfg answer ;)

Reply:
> As I understood the description, it should happen only during a short time.

This is correct. Hue battery devices communicate their status to the Hue bridge by means of two processes..
1. event driven: if someone presses a button on such a device, it immediately sends its new status to the bridge.
2. polling: in case nobody presses a button, the bridge also does a slow polling of the battery state; the polling interval is not too fast in order to prevent waking the device too often

This bug would only occur if a new device is added to the Hue bridge and then immediately also added to OH before either case 1. or 2. will have occurred. Prior to this PR that case would cause an NPE, whereas after this PR the battery state is reported as critical until either case 1. or 2. actually occurs, at which time it is updated to the new actual value.

This is quite a seldom case that rights itself quite quickly, so IMHO temporarily reporting critical is actually perfectly Ok.